packages/utils/src/__tests__/joiner/remote-joiner.ts

```diff
@@ -13,10 +14,37 @@ const container = {
   },
 } as MedusaContainer
 
+const fetchServiceDataCallback = async (
```

Comment:
I like the idea, tell me if I am wrong, but in that case the remote joiner will just prepare everything needed, by default we have our internal callback but the user could manage the call to his service manually by giving a custom call back right?